### ISSUE TITLE
Style/SingleLineMethods should not correct to endless methods when Style/EndlessMethod is disabled

### ISSUE THREAD

**Issue by searls** [rubocop/rubocop#11899]
**This is just a question and came up when researching https://github.com/standardrb/standard/issues/489**

I can't help but wonder if the guard clause's boolean logic in [this method](https://github.com/rubocop/rubocop/blob/master/lib/rubocop/cop/style/single_line_methods.rb#L138) in `Style/SingleLineMethods` isn't reversed.

This method is used in determining if a single-line method should be converted to an endless method:

```ruby
def disallow_endless_method_style?
  endless_method_config = config.for_cop('Style/EndlessMethod')
  return false unless endless_method_config['Enabled']
  
  endless_method_config['EnforcedStyle'] == 'disallow'
end
```

Converted to English and with the double-negatives unwound, the guard clause seems to be saying "If Style/EndlessMethod is disabled, then *allow* violations of this cop to be converted to Endless methods".  Am I reading that wrong, or is that logically backwards? If someone has disabled `Style/EndlessMethod`, shouldn't endless method corrections by this cop _be disallowed_?

I expected the line to read:

```ruby
  return true unless endless_method_config['Enabled']
```

Do I have this right? Just a quick spot-check, as I may be totally wrong here—it's a lot of boolean negation to keep straight, and I'm not sure how the two cops interact

**Comment by maxjacobson** [comment]
I'm here to mount a mild defense of the current behavior.

When someone has this configuration:

```yaml
AllCops:
  TargetRubyVersion: 3.2.2

Style/EndlessMethod:
  Enabled: false

Style/SingleLineMethods:
  Enabled: true
```

They haven't explicitly disallowed endless methods in their codebase. If they had, it would look like:


```yaml
AllCops:
  TargetRubyVersion: 3.2.2

Style/EndlessMethod:
  Enabled: true
  EnforcedStyle: disallow

Style/SingleLineMethods:
  Enabled: true
```

Given that they haven't disallowed endless methods, and they're targeting a ruby that supports endless methods, why shouldn't `Style/SingleLineMethods` introduce an endless method?

Note: if you're persuaded by this argument, the docs and test coverage should be updated to reflect that this is expected behavior.

**Comment by koic** [comment]
This is about the default behavior of autocorrection `Style/SingleLineMethods`. It does not deny the use of endless method definition. And this fix conforms to the behavior documented below:

> If `Style/EndlessMethod` is enabled with `EnforcedStyle: allow_single_line` or
> `allow_always`, single-line methods will be autocorrected to endless
> methods if there is only one statement in the body.

https://docs.rubocop.org/rubocop/1.51/cops_style.html#stylesinglelinemethods

**Comment by searls** [comment]
Thanks, @koic!

**Comment by koic** [comment]
You are welcome, @searls!
